Client/Frontend/Home/ActivityStreamTopSitesCell.swift

```diff
@@ -0,0 +1,393 @@
+    }()
+
+    lazy private var titleLabel: UILabel = {
+        let titleLabel = UILabel()
+        titleLabel.layer.masksToBounds = true
+        titleLabel.textAlignment = .Center
+        titleLabel.font = TopSiteCellUX.TitleFont
+        titleLabel.textColor = TopSiteCellUX.TitleTextColor
+        titleLabel.backgroundColor = TopSiteCellUX.TitleBackgroundColor
+        return titleLabel
+    }()
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        contentView.layer.cornerRadius = TopSiteCellUX.CellCornerRadius
+        contentView.layer.masksToBounds = true
+
+        contentView.addSubview(titleLabel)
+        contentView.addSubview(imageView)
+
+        let heightInset = frame.height * TopSiteCellUX.TitleInsetPercent
+        titleLabel.snp_makeConstraints { make in
+            make.edges.equalTo(self).inset(UIEdgeInsetsMake(heightInset, 0, 0, 0))
+        }
+
+        imageView.snp_makeConstraints { make in
+            make.size.equalTo(CGSize(width: self.frame.width/2, height: self.frame.height/2))
+
+            // Add an offset to the image to make it appear centered with the titleLabel
+            let offset = self.frame.height - CGFloat(heightInset)
+            make.center.equalTo(self.snp_center).offset(UIEdgeInsets(top: -offset/2, left: 0, bottom: 0, right: 0))
+        }
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func prepareForReuse() {
+        contentView.backgroundColor = UIColor.lightGrayColor()
+        imageView.image = nil
+        titleLabel.text = ""
+    }
+
+    private func setImageWithURL(url: NSURL) {
+        imageView.sd_setImageWithURL(url) { [unowned self] (img, err, type, url) -> Void in
+            guard let img = img else {
+                self.imageView.image = FaviconFetcher.getDefaultFavicon(url)
+                self.contentView.backgroundColor = UIColor.lightGrayColor()
+                return
+            }
+
+            // Get dominant colors using a scaled 25/25 image.
+            img.getColors(CGSize(width: 25, height: 25)) { colors in
+                //In cases where the background is white. Force the background color to a different color
+                var bgColor: UIColor
+                if colors.backgroundColor.isWhite {
+                    let colorArr = [colors.detailColor, colors.primaryColor].filter { !$0.isWhite }
+                    if colorArr.isEmpty {
+                        bgColor = UIColor.lightGrayColor()
+                    } else {
+                        bgColor = colorArr.first!
+                    }
+                } else {
+                    bgColor = colors.backgroundColor
+                }
+                self.contentView.backgroundColor = bgColor
+            }
+        }
+    }
+
+    func configureWithTopSiteItem(site: TopSiteItem) {
+        titleLabel.text = site.urlTitle
+        if let favURL = site.faviconURL {
```

Comment:
Nit: Would read better as a `guard` since you want the favicon to normally be there but otherwise, just use the default favicon.